From 9f4e68261b89df87457f46999c471d11754b5de2 Mon Sep 17 00:00:00 2001
From: Lech Perczak <lech.perczak@gmail.com>
Date: Sun, 6 Oct 2024 15:57:44 +0200
Subject: [PATCH] bcm53xx: dts: meraki-mx6x: add OpenWrt-specific LED aliases

Assign green power LED for "boot", "running", orange for "upgrade" and
red for "failsafe" functions - the same as done for MR33 and MR74.

Signed-off-by: Lech Perczak <lech.perczak@gmail.com>
---
 arch/arm/boot/dts/broadcom/bcm958625-meraki-alamo.dtsi   | 9 ++++++++-
 arch/arm/boot/dts/broadcom/bcm958625-meraki-kingpin.dtsi | 8 +++++++-
 .../boot/dts/broadcom/bcm958625-meraki-mx6x-common.dtsi  | 4 ++--
 3 files changed, 17 insertions(+), 4 deletions(-)

--- a/arch/arm/boot/dts/broadcom/bcm958625-meraki-alamo.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm958625-meraki-alamo.dtsi
@@ -8,6 +8,13 @@
 #include "bcm958625-meraki-mx6x-common.dtsi"
 
 / {
+	aliases {
+		led-boot = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_fault;
+		led-failsafe = &led_failsafe;
+	};
+
 	keys {
 		compatible = "gpio-keys-polled";
 		autorepeat;
@@ -55,7 +62,7 @@
 			gpios = <&gpioa 26 GPIO_ACTIVE_LOW>;
 		};
 
-		led-4 {
+		led_fault: led-4 {
 			/* amber:power */
 			function = LED_FUNCTION_FAULT;
 			color = <LED_COLOR_ID_AMBER>;
--- a/arch/arm/boot/dts/broadcom/bcm958625-meraki-kingpin.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm958625-meraki-kingpin.dtsi
@@ -8,6 +8,12 @@
 #include "bcm958625-meraki-mx6x-common.dtsi"
 
 / {
+	aliases {
+		led-boot = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_fault;
+		led-failsafe = &led_failsafe;
+	};
 
 	keys {
 		compatible = "gpio-keys-polled";
@@ -104,7 +110,7 @@
 			gpios = <&gpioa 29 GPIO_ACTIVE_LOW>;
 		};
 
-		led-a {
+		led_fault: led-a {
 			/* amber:power */
 			function = LED_FUNCTION_FAULT;
 			color = <LED_COLOR_ID_AMBER>;
--- a/arch/arm/boot/dts/broadcom/bcm958625-meraki-mx6x-common.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm958625-meraki-mx6x-common.dtsi
@@ -14,14 +14,14 @@
 	pwm-leds {
 		compatible = "pwm-leds";
 
-		led-1 {
+		led_failsafe: led-1 {
 			function = LED_FUNCTION_INDICATOR;
 			color = <LED_COLOR_ID_RED>;
 			pwms = <&pwm 1 50000>;
 			max-brightness = <255>;
 		};
 
-		led-2 {
+		led_power: led-2 {
 			function = LED_FUNCTION_POWER;
 			color = <LED_COLOR_ID_GREEN>;
 			pwms = <&pwm 2 50000>;
